### PR TITLE
Add Fabric namespace support

### PR DESCRIPTION
**Pull request by niravcodes**

### CODE REVIEW

File: demo/video/package.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
     "preview": "vite preview"
   },
   "dependencies": {
-    "@signalwire-community/react": "*",
-    "@signalwire/js": "^3.17.0",
+    "@signalwire-community/react": "1.4.2-dev.1",
```
Comment (danieleds) [inline demo/video/package.json:11]: should this remain `"*"`? It was needed for Lerna if I remember correctly

File: demo/video/src/Fabric/DemoFabric.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { useEffect, useState } from "react";
+
+function DemoFabric() {
+  const client = Fabric.useFabric(import.meta.env.VITE_FABRIC_TOKEN);
+  const addresses = Fabric.useAddresses(client, 10000);
+
+  return (
+    <div>
+      {import.meta.env.VITE_FABRIC_TOKEN && client && addresses !== null ? (
+        <Fabric.Video client={client} address={addresses[5]} />
```
Comment (danieleds) [inline demo/video/src/Fabric/DemoFabric.tsx:11]: instead of joining `addresses[5]`, can we filter for `room` address types and join the first of them?

File: packages/react-native/src/components/Fabric/Video.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import RemoteStream from '../RemoteStream';
+registerGlobals();
+
+interface IVideoNativeProps extends Fabric.IVideoProps {
+  style: any;
+}
+
+export function Video({ style, onRoomReady, ...props }: IVideoNativeProps) {
+  const [roomSession, setRoomSession] =
+    useState<SignalWire.Video.RoomSession | null>(null);
```
Comment (danieleds) [inline packages/react-native/src/components/Fabric/Video.tsx:16]: just a note: this works for now, but the type will not be `RoomSession` anymore, eventually. But let's keep it like this for now.

File: packages/react-native/src/components/Fabric/Video.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    <Fabric.CoreVideo
+      {...props}
+      onRoomReady={(r: SignalWire.Video.RoomSession) => {
+        setRoomSession(r);
+        onRoomReady?.(r);
+      }}
+      audio={props.audio ?? true}
+      video={props.video ?? true}
+    >
+      <RemoteStream roomSession={roomSession ?? undefined} style={style} />
```
Comment (danieleds) [inline packages/react-native/src/components/Fabric/Video.tsx:28]: note: this component will need to be updated to accept either a roomSession or whatever type of objects the Fabric namespace will produce. Let's keep it like this for now.

File: packages/react/src/Fabric/CoreVideo.tsx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+        // @ts-expect-error undefined is not assignable to rootElement
+        rootElement: props.rootElement?.current ?? undefined,
+      });
+      roomSessionRef.current = currentRoomSession;
+      setRoomSession(currentRoomSession)
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      currentRoomSession.on("memberList.updated", () => {}); // Workaround for cloud-product/4681 (internal)
+      props.onRoomReady?.(currentRoomSession);
+      await currentRoomSession.join();
```
Comment (danieleds) [inline packages/react/src/Fabric/CoreVideo.tsx:70]: the `join` method will probably be removed. Do we need it?

File: packages/react/src/Fabric/useAddresses.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { Fabric } from "@signalwire/js";
+import { useEffect, useState } from "react";
+
+/**
+ * Given a client, poll the addresses available.
+ * @param client is the fabric client
+ * @param delay in ms, the polling time
+ * @returns null | address[]
+ */
+export function useAddresses(client: Fabric.Client | null, delay = 10000) {
```
Comment (danieleds) [inline packages/react/src/Fabric/useAddresses.tsx:10]: we should hardcode the delay (10,000 is fine) to make sure that the interface won't change once this is implemented with the proper apis

File: packages/react/tsconfig.build.esm.json [hunk 7]
```diff
@@ -3,7 +3,7 @@
   "compilerOptions": {
     "module": "ES2020",
     "outDir": "dist/esm",
-    "declaration": false
+    "declaration": true
```
Comment (danieleds) [inline packages/react/tsconfig.build.esm.json:6]: why is this needed?

**Comment by danieleds** [comment]
@niravcodes the `unregisterDevice` should automatically use the id that is returned by `registerDevice`. We'll need to store it after calling `registerDevice`

File: packages/react-native/package.json [hunk 8]
```diff
@@ -64,7 +64,7 @@
     "react-native": "0.68.2",
     "react-native-builder-bob": "^0.18.3",
     "react-native-get-random-values": "^1.8.0",
-    "react-native-webrtc": "^106.0.7",
+    "react-native-webrtc": ">=106.0.0",
```
Comment (danieleds) [inline packages/react-native/package.json:67]: Do we want this? Asking since the config-plugin is only explicitly compatible with v106.x.x

**Comment by danieleds** [comment]
@niravcodes it seems CI is failing after the version upgrade? could you check?

File: packages/react-native/package.json (hunk 8 above)
Comment (niravcodes) [reply packages/react-native/package.json:67]: The config plugin does work fine with newer versions of react-native-webrtc up till v111.

I wanted to maximize compatible versions, even if you're not using expo. I could cap it between 106 and 111...

**Comment by niravcodes** [comment]
@danieleds Fixing the CI build issue. It is caused by the demo failing to build because of version mismatch, so no issues in the core packages.

File: packages/react-native/package.json (hunk 8 above)
Comment (danieleds) [reply packages/react-native/package.json:67]: > I could cap it between 106 and 111

@niravcodes I like that

Comment (niravcodes) [reply packages/react-native/package.json]: ok done!

File: packages/react/src/SignalWire/useSignalWire.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  useEffect(() => {
+    if (token === undefined) return;
+    async function createClient() {
+      const _client = await SignalWire({
+        token: token,
+      });
+      setClient(_client);
+    }
+    createClient();
+  }, [token]);
```
Comment (danieleds) [inline packages/react/src/SignalWire/useSignalWire.ts:25]: @niravcodes should we call `client.disconnect()` in the cleanup fn?

File: packages/react/src/SignalWire/CoreVideo.tsx [hunk 10]
```diff
@@ -56,57 +55,49 @@ export function CoreVideo({ ...props }: IVideoProps) {
...
 
-      const currentRoomSession = await props.client.createCall({
-        uri: props.address.channels.video,
+      console.log("Dialing to the address");
+      const currentCall = await props.client.dial({
+        to: props.address.channels.video,
 
         // @ts-expect-error undefined is not assignable to rootElement
-        rootElement: props.rootElement?.current ?? undefined,
+        // rootElement: props.rootElement?.current ?? undefined,
```
Comment (danieleds) [inline packages/react/src/SignalWire/CoreVideo.tsx:63]: @niravcodes just to understand, why do we need to comment this out?

**Comment by niravcodes** [comment]
> @niravcodes did you make manual changes to the `demo/pushNotifications/android/` and `demo/pushNotifications/ios/` folders?

Yup! Unfortunately this is one of those problems that needs manual work on the native side. I'm working on a set of notes to make it as easy as possible but it's still somewhat messy.

File: packages/react/src/SignalWire/CoreVideo.tsx (hunk 10 above)
Comment (niravcodes) [reply packages/react/src/SignalWire/CoreVideo.tsx:63]: The new SignalWire interface expects rootElement to be registered at the time of instantiation and not during dial so we dont actually need the line there.

**Review by danieleds — APPROVED** [review]
